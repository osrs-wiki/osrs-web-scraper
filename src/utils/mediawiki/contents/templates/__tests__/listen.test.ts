import ListenTemplate, { ListeTemplateOptions } from "../listen";

describe("ListenTemplate", () => {
  test("it should render with only a file", () => {
    expect(new ListenTemplate("test.mp3").build()).toMatchSnapshot();
  });

  test.each<ListeTemplateOptions>([
    { align: "left" },
    { title: "test title" },
    { align: "left", title: "test title" },
  ])("it should render with options: %j", (options) => {
    expect(new ListenTemplate("test.mp3", options).build()).toMatchSnapshot();
  });
});
