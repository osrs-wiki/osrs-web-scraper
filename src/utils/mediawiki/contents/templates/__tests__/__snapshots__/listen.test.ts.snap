// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListenTemplate it should render with only a file 1`] = `
MediaWikiTemplate {
  "children": undefined,
  "name": "Listen",
  "params": Array [
    Object {
      "key": "filename",
      "value": "test.mp3",
    },
  ],
}
`;

exports[`ListenTemplate it should render with options: {"align":"left","title":"test title"} 1`] = `
MediaWikiTemplate {
  "children": undefined,
  "name": "Listen",
  "params": Array [
    Object {
      "key": "filename",
      "value": "test.mp3",
    },
    Object {
      "key": "align",
      "value": "left",
    },
    Object {
      "key": "title",
      "value": "test title",
    },
  ],
}
`;

exports[`ListenTemplate it should render with options: {"align":"left"} 1`] = `
MediaWikiTemplate {
  "children": undefined,
  "name": "Listen",
  "params": Array [
    Object {
      "key": "filename",
      "value": "test.mp3",
    },
    Object {
      "key": "align",
      "value": "left",
    },
  ],
}
`;

exports[`ListenTemplate it should render with options: {"title":"test title"} 1`] = `
MediaWikiTemplate {
  "children": undefined,
  "name": "Listen",
  "params": Array [
    Object {
      "key": "filename",
      "value": "test.mp3",
    },
    Object {
      "key": "title",
      "value": "test title",
    },
  ],
}
`;
